Index: lib/vendor/symfony/lib/plugins/sfDoctrinePlugin/lib/vendor/doctrine/Doctrine/Connection/UnitOfWork.php
===================================================================
--- lib/vendor/symfony/lib/plugins/sfDoctrinePlugin/lib/vendor/doctrine/Doctrine/Connection/UnitOfWork.php	(revision 2493)
+++ lib/vendor/symfony/lib/plugins/sfDoctrinePlugin/lib/vendor/doctrine/Doctrine/Connection/UnitOfWork.php	(working copy)
@@ -773,6 +773,19 @@
                 }
             }
         }
+        
+        /* This hack puts UllUser at the first position
+         * Reason: Every UllRecord has a relation to UllUser in the form
+         * of Creator and Updator.
+         * 
+         * UllUser has a lot of local key relations itself (e.g. UllLocation)
+         * It still works, because the constraints for UllUser
+         * are not exported to the database because Mysql 5 cannot delete
+         * records with self constraints 
+         */
+        $UllUserIndex = array_search('UllUser', $flushList);
+        unset($flushList[$UllUserIndex]);
+        array_unshift($flushList, 'UllUser');
 
         return array_values($flushList);
     }
