Index: lib/vendor/symfony/lib/helper/TextHelper.php
===================================================================
--- lib/vendor/symfony/lib/helper/TextHelper.php	(revision 2765)
+++ lib/vendor/symfony/lib/helper/TextHelper.php	(working copy)
@@ -283,5 +283,10 @@
  */
 function _auto_link_email_addresses($text)
 {
-  return preg_replace('/([\w\.!#\$%\-+.]+@[A-Za-z0-9\-]+(\.[A-Za-z0-9\-]+)+)/', '<a href="mailto:\\1">\\1</a>', $text);
+  // Taken from http://snippets.dzone.com/posts/show/6156
+  return preg_replace("#(^|[\n ])([a-z0-9&\-_\.]+?)@([\w\-]+\.([\w\-\.]+\.)*[\w]+)#i", "\\1<a href=\"mailto:\\2@\\3\">\\2@\\3</a>", $text);
+  
+  // Removed since it destroys already linked emails 
+  // Example:   <a href="mailto:me@example.com">bar</a> gets <a href="mailto:me@example.com">bar</a> gets <a href="mailto:<a href="mailto:me@example.com">bar</a>
+  //return preg_replace('/([\w\.!#\$%\-+.]+@[A-Za-z0-9\-]+(\.[A-Za-z0-9\-]+)+)/', '<a href="mailto:\\1">\\1</a>', $text);
 }
